sdk/storage/azure-storage-blob/azure/storage/blob/_download.py

```diff
@@ -739,7 +745,7 @@ def read(self, size: int = -1, *, chars: Optional[int] = None) -> T:
         if not self._text_mode and self._encoding:
             try:
                 # This is technically incorrect to do, but we have it for backwards compatibility.
-                data = data.decode(self._encoding)
+                data = cast(bytes, data).decode(self._encoding)
```

Comment:
I can't put a comment on the line, but the end of this function (typehinted for `T`) also has this issue:
azure\storage\blob\_download.py:752: error: Incompatible return value type (got "Union[bytes, str]", expected "bytes")  [return-value]
azure\storage\blob\_download.py:752: error: Incompatible return value type (got "Union[bytes, str]", expected "str")  [return-value]

Reply:
Ignore it